File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        long long res = 1;
        int cnt =0;
        vector<int> ress;
        for (int i =0 ; i<nums.size(); i++) {
            if (nums[i] != 0) res *= nums[i];
            else cnt++;
        }
        for (int i =0 ; i<nums.size(); i++) {
            if (nums[i] != 0) ress.push_back(cnt?0: res / nums[i]);
            else ress.push_back(cnt>1 ?0 : res);
        }
        return ress;
    }
};

File: Data Structures & Algorithms/majority-element/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        dct = {}
        ans = nums[0]
        for i in nums:
            dct[i] = 1 + dct.get( i , 0)
            if dct[i] >= dct[ans] and dct[i] >=  len(nums) // 2:
                ans = i
        return ans          